plugins/select_colorscheme.lua

```diff
@@ -0,0 +1,135 @@
+-- mod-version:2 -- lite-xl 2.0
+local core = require "core"
+local command = require "core.command"
+local common = require "core.common"
+local style = require "core.style"
+local CommandView = require "core.commandview"
+
+-- ----------------------------------------------------------------
+local PATH_CONFIG = USERDIR .. "/color_settings.lua"
+
+local Settings = {}
+Settings.color_scheme = ""
+Settings.color_list = {}
+local color_default = {name = "Default", module = "core.style"}
+local plugin_enable = false
+
+--  =========================Proxy method==========================
+local move_suggestion_idx = CommandView.move_suggestion_idx
+
+function CommandView:move_suggestion_idx(dir)
+  move_suggestion_idx(self, dir)
+  if plugin_enable then
+    local color_name = self.suggestions[self.suggestion_idx].text
+    Settings:change_color(color_name)
+  end
+end
+
+local on_quit_project = core.on_quit_project
+
+function core.on_quit_project()
+  Settings:save_settings()
+  on_quit_project()
+end
+-- ----------------------------------------------------------------
+
+function Settings:get_color_list()
+  return self.color_list
+end
+
+function Settings:init()
+  self:load_settings()
+  self:make_color_list()
+end
+
+function Settings:make_color_list()
+  for _, root_dir in ipairs {DATADIR, USERDIR} do
+    local plugin_dir = root_dir .. "/colors"
+    for _, filename in ipairs(system.list_dir(plugin_dir) or {}) do
+      table.insert(self.color_list, filename:match("(.-)%.lua$"))
+    end
+  end
+  table.insert(self.color_list, color_default.name)
+end
+
+function Settings:is_change_color(color_name)
+  return not (self.color_scheme == color_name)
+end
+
+function Settings:get_color_scheme()
+  return (self.color_scheme == "") and color_default.name or self.color_scheme
+end
+
+local function make_color_module_name(name)
+  return (name == color_default.name) and color_default.module or "colors."..name  
+end
+
+function Settings:change_color(name)
+    if self:is_change_color(name) then
+      core.reload_module(make_color_module_name(name))
+      self.color_scheme = name
+    end
+end
+
+function Settings:save_settings()
+  local fp = io.open(PATH_CONFIG, "w")
+  if fp then
+    fp:write(self.color_scheme)
+    fp:close()
+  end
+end
+
+function Settings:load_settings()
+  local fp = io.open(PATH_CONFIG, "r")
+  if fp then
+    local name = fp:read("*a")
+    core.reload_module(make_color_module_name(name))
+    Settings.color_scheme = name
+    fp:close()
+  end
+end
+
+-- -------------------------------Utility--------------------------
+local function table_remove_value(list, value)
+  local remove_idx = nil
+  for i,v in ipairs(list) do
+    if v == value then
+      remove_idx = i
+      break
+    end
+  end
+  if remove_idx then
+    table.remove(list, remove_idx)
+  end
+  return list
+end
```

Comment:
That's fine but I proposed a shorter, simplified version. There is a reason to keep it that way?

Reply:
> That's fine but I proposed a shorter, simplified version. There is a reason to keep it that way?

I agree with you. my mistake